pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -22,46 +22,60 @@ func createVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool, strin
 		log.Infof("createVolume(%s) from container %s", status.Key(), status.ReferenceName)
 		return createContainerVolume(ctx, status, status.ReferenceName)
 	}
-	log.Infof("createVolume(%s) from disk %s", status.Key(), status.FileLocation)
-	return createVdiskVolume(ctx, status, status.FileLocation)
+	log.Infof("createVolume(%s) from disk %s", status.Key(), status.ReferenceName)
+	return createVdiskVolume(ctx, status, status.ReferenceName)
 }
 
 // createVdiskVolume does not update status but returns
 // new values for VolumeCreated, FileLocation, and error
 func createVdiskVolume(ctx *volumemgrContext, status types.VolumeStatus,
-	srcLocation string) (bool, string, error) {
+	ref string) (bool, string, error) {
 
 	created := false
-	if status.ReadOnly {
-		log.Infof("createVolume(%s) ReadOnly", status.Key())
-		created = true // To make doUpdate proceed
-		return created, srcLocation, nil
-	}
 
+	// this is the target location, where we expect the volume to be
 	filelocation := status.PathName()
 	if _, err := os.Stat(filelocation); err == nil {
 		errStr := fmt.Sprintf("Can not create %s for %s: exists",
 			filelocation, status.Key())
 		log.Error(errStr)
-		return created, srcLocation, errors.New(errStr)
+		return created, ref, errors.New(errStr)
 	}
-	log.Infof("Copy from %s to %s", srcLocation, filelocation)
-	created = true // So we will delete later even if partial failure
-	if err := cp(filelocation, srcLocation); err != nil {
-		errStr := fmt.Sprintf("Copy failed from %s to %s: %s\n",
-			srcLocation, filelocation, err)
+
+	// use the edge-containers library to extract the data we need
+	puller := registry.Puller{
+		Image: ref,
+	}
+	resolver, err := ctx.casClient.Resolver()
+	if err != nil {
+		errStr := fmt.Sprintf("error getting CAS resolver: %v", err)
 		log.Error(errStr)
-		return created, filelocation, errors.New(errStr)
+		return created, ref, errors.New(errStr)
 	}
-	// Do we need to expand disk?
-	err := maybeResizeDisk(filelocation, status.MaxVolSize)
+
+	_, artifact, err := puller.Pull(filelocation, false, os.Stderr, resolver)
```

Comment:
How is os.Stderr used here? If it is for logging we should instead pass something which makes the written stuff go to logrus.

Reply:
It actually isn't. The `puller.Pull()` only uses the third arg (`io.Writer`) if the second arg (`bool`) is set to `true`. If it ism it prints out lots of logs.

To be fair, that part probably should be reworked. It would be nice if `log` were an interface, or fulfilled a known interface, and then you can just pass `logrus` or `log` or anything.

Since it is false now, it won't log anything (short of returning an `error` if there is an issue), so let's leave it for now. I will try to think of a better way to handle this in the underlying library after this is done.